test/unit/apk-signing-specs.js

```diff
@@ -0,0 +1,146 @@
+      password = 'android',
+      selendroidTestAppPackage = 'io.selendroid.testapp';
+
+describe('signing', async () => {
+  let adb = new ADB(),
+      withExecMock = (fn) => {
+        return () => {
+          let mocks = {};
+          before(async () => {
+            await adb.createADB();
+            adb.keystorePath = keystorePath;
+            adb.keyAlias = keyAlias;
+            adb.keystorePassword = password;
+            adb.keyPassword = password;
+            mocks.teen_process = sinon.mock(teen_process);
+            mocks.adb = sinon.mock(adb);
+            mocks.utils = sinon.mock(utils);
+            mocks.tempDir = sinon.mock(tempDir);
+          });
+          after(() => {
+            mocks.teen_process.restore();
+            mocks.adb.restore();
+            mocks.utils.restore();
+            mocks.tempDir.restore();
+          });
+          fn(mocks);
+        };
+      };
+  describe('signWithDefaultCert', withExecMock((mocks) => {
+    it('should call exec with correct args', async () => {
+      let signPath = path.resolve(helperJarPath, 'sign.jar');
+      mocks.teen_process.expects("exec")
+        .once().withExactArgs(java, ['-jar', signPath, selendroidTestApp, '--override'])
+        .returns("");
+      (await adb.signWithDefaultCert(selendroidTestApp));
+      mocks.teen_process.verify();
+    });
+    it('should throw error for invalid file path', async () => {
+      let dummyPath = "dummyPath";
+      await adb.signWithDefaultCert(dummyPath).should.eventually.be.rejected;
+      mocks.teen_process.verify();
+    });
+  }));
+  describe('signWithCustomCert', withExecMock((mocks) => {
+    it('should call exec with correct args', async () => {
+      let jarsigner = path.resolve(process.env.JAVA_HOME, 'bin', 'jarsigner');
+      adb.useKeystore = true;
+      mocks.teen_process.expects("exec")
+        .withExactArgs(java, ['-jar', path.resolve(helperJarPath, 'unsign.jar'), selendroidTestApp])
+        .returns("");
+      mocks.teen_process.expects("exec")
+        .withExactArgs(jarsigner, ['-sigalg', 'MD5withRSA', '-digestalg', 'SHA1',
+                                   '-keystore', keystorePath, '-storepass', password,
+                                   '-keypass', password, selendroidTestApp, keyAlias])
+        .returns("");
+      (await adb.signWithCustomCert(selendroidTestApp));
+      mocks.teen_process.verify();
+    });
+  }));
+  describe('getKeystoreMd5', withExecMock((mocks) => {
+    it('should call exec with correct args', async () => {
+      let h = "a-fA-F0-9";
+      let keytool = path.resolve(process.env.JAVA_HOME, 'bin', 'keytool');
+      let md5Str = ['.*MD5.*((?:[', h, ']{2}:){15}[', h, ']{2})'].join('');
+      let md5 = new RegExp(md5Str, 'mi');
+      adb.useKeystore = true;
+      mocks.teen_process.expects("exec")
+        .once().withExactArgs(keytool, ['-v', '-list', '-alias', keyAlias,
+                            '-keystore', keystorePath, '-storepass',
```

Comment:
align this under the array bracket, or just create the args array as a variable before using it